src/main/java/org/jboss/threads/EnhancedQueueExecutor.java

```diff
@@ -1519,8 +1520,10 @@ private QNode getOrAddNode() {
                         return taskNode;
                     }
                 } else if (headNext instanceof PoolThreadNode || headNext == null) {
-                    PoolThreadNode newNode;
-                    newNode = new PoolThreadNode((PoolThreadNode) headNext, Thread.currentThread());
+                    if (newNode == null) {
+                        newNode = new PoolThreadNode(null, Thread.currentThread());
```

Comment:
This constructor is only called here IIRC, so the first parameter can be safely dropped instead of always being `null`.

Reply:
Good call. It turns out this was the only constructor for any subtype of QNode that used the QNode(QNode next) ctor with a non-null value, so I've removed that entirely as well.